.github/workflows/CI.yaml

```diff
@@ -1,19 +1,69 @@
 name: CI
 
 on:
-  pull_request:
   push:
 
 jobs:
 
-  nix-build:
-    name: Nix build
+  nix-build-docs:
+    name: Nix build docs
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2.3.4
       - uses: cachix/install-nix-action@v17
       - name: Build all packages
         run: nix build -L .#docs
-      # - name: Build devShell
-      #   run: nix build -L '.#devShells.x86_64-linux.hacking-on-purenix' '.#devShells.x86_64-linux.use-purenix'
 
+  nix-build-server:
+    name: Nix build server
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: cachix/install-nix-action@v16
+      - name: Build binplz-server
+        run: nix build -L '.#server'
+      - name: Build binplz-server shell
+        run: nix build -L '.#devShell.x86_64-linux'
+
+  stack-build:
+    name: Stack ${{ matrix.resolver }} on ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        resolver: [ nightly, lts-19, lts-18, lts-17 ]
+        os: [ macos-latest, ubuntu-latest ]
```

Comment:
We could potentially take out most of these in order to make CI run faster.

I guess we could even get rid of the whole `stack-build` and `cabal-build` jobs, since we'll probably be deploying the binaries after doing `nix build`.

Reply:
@jonascarpay You could take out `macos` as well, since we'll likely never be able to run this server on darwin (since you can't statically link stuff on darwin).

Reply:
I trimmed it down a bit in 9c967fdcdb89cf7c6366c42eb2306fab7d6cdb7a

I still like having both cabal and stack, since as you can see in https://github.com/binplz/binplz.dev/actions/runs/2798715131, occasionally one catches issues that the others might not.

I'll spend some time at some point to properly figure out how to robustly cache stack artifacts, that should help build times a lot.

Reply:
> You could take out macos as well, since we'll likely never be able to run this server on darwin (since you can't statically link stuff on darwin).

Oh, I didn't know, good point. Can you not statically link _on_ darwin or _for_ darwin (or both)?

Reply:
> Oh, I didn't know, good point. Can you not statically link on darwin or for darwin (or both)?

My understanding is that static-linking is still a second-class citizen most places (even in Nixpkgs).  In the Linux world, there are a whole lot of packages that are able to be successfully compiled while dynamically linking to glibc.  Only a subset of them are able to be compiled while dynamically linking to musl (because musl doesn't provide exactly the same interface as glibc in some places??).  And then only a subset of them are able to be statically-linked to musl (for whatever reason).

In the darwin world, there are problems with static-linking _anything_.  If I remember correctly, the problem is that like the kernel (or maybe darwin libc?) doesn't have a stable syscall interface or something?  So even if you were able to statically-link a binary on darwin, you would only be able to use it with your current libc/kernel version.  It kind of defeats the purpose of static-linking.  Because of that, no one has put any effort to getting static-linking working on darwin.  

I'm not super familiar with this, so I could be getting some of the details wrong.  But I am pretty sure that Nixpkgs doesn't really have any support for static-linking on darwin.

Reply:
Interesting, I didn't know! Then, yeah, let's ditch macos builds, they're super slow, and unlike caba/stack/nix, I don't think I've ever found an error by building on macos that was somehow more generally useful to know.

Reply:
f176e64b38bbfc21b5e147f1d99180a1efdef045